examples/chat.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::sync::{Arc, Mutex};
+
+use futures::{sink::SinkExt, stream::StreamExt};
+
+use tokio::sync::broadcast;
+
+use axum::handler::get;
+use axum::response::Html;
+use axum::routing::RoutingDsl;
+use axum::ws::{ws, Message, WebSocket};
+use axum::{extract, route, AddExtensionLayer};
+
+// Our shared state
+struct AppState {
+    user_map: Mutex<HashMap<String, ()>>,
```

Comment:
What do you think about using `HashSet<String>` instead of `HashMap<String, ()>`? They're basically the same.